src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/ServiceDescriptionsApiController.java

```diff
@@ -191,12 +191,13 @@ public ServiceDescriptionsApiController(ServiceDescriptionService serviceDescrip
                             ignoreWarnings.getIgnoreWarnings()));
         } catch (WsdlParser.WsdlNotFoundException | UnhandledWarningsException
                 | InvalidUrlException | InvalidWsdlException
-                | ServiceDescriptionService.WrongServiceDescriptionTypeException e) {
+                | ServiceDescriptionService.WrongServiceDescriptionTypeException
+                | OpenApiParser.ParsingException e) {
             throw new BadRequestException(e);
         } catch (ServiceDescriptionService.ServiceAlreadyExistsException
                 | ServiceDescriptionService.WsdlUrlAlreadyExistsException e) {
             throw new ConflictException(e);
-        } catch (ServiceDescriptionNotFoundException e) {
+        } catch (ServiceNotFoundException | ServiceDescriptionNotFoundException e) {
```

Comment:
ServiceDescriptionException is not really a HTTP 404 resource not found. API URL defines what the _resource_ is, in this case `/service-descriptions/{id}/refresh`, and that does exist. If the processing of this request fails then error is something else, like for example http 500 internal server error, http 409 conflict, or something else.

I am not 100% sure if `ServiceNotFoundException` is correct exception class to be used in this case altogether. `refreshRESTServiceDescription` does not throw it even though it claims to. `refreshOPENAPI3ServiceDescription` throws it if we had for some reason a service description without any services. So it is more a problem with a service description, not a problem of _not finding one specific service_. It could have a different exception type, or just be a (DeviationAware)RuntimeException. 

Or keep it as a ServiceNotFoundException (which is not 100% accurate IMO) but at least map it to http 500 internal server error? Isn't that what it is, something weird has happened for it to be possible, and it is not anything that client request has caused, so it cannot be http 4xx?

Reply:
Pushed a commit changing the exception to DeviationAwareRuntimeException in a few places related to this.